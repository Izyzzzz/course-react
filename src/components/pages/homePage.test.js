import React from 'react';
import HomePage from './homePage';
import {shallow} from 'enzyme';

describe('Testing <HomePage/>', () => {
    const home = shallow(<HomePage/>);
    describe('Testing snap & state', () => {
        it('HomePage have rendered correctly', () => {        
            expect(home).toMatchSnapshot();
        });
        it('HomePage state "isEdit" is false', () => {
            expect(home.state().isEdit).toBeFalsy();
        });
    });
});