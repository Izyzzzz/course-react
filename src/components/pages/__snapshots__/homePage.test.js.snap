// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <HomePage/> Testing snap & state HomePage have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomePage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Col
        lg={
          Object {
            "offset": 3,
            "size": 6,
          }
        }
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <RandomChar />
        <button
          className="button-hide d-flex justify-content-center"
          onClick={[Function]}
        >
          Hide
        </button>
      </Col>,
      "tag": "div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <RandomChar />,
          <button
            className="button-hide d-flex justify-content-center"
            onClick={[Function]}
          >
            Hide
          </button>,
        ],
        "lg": Object {
          "offset": 3,
          "size": 6,
        },
        "tag": "div",
        "widths": Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hide",
            "className": "button-hide d-flex justify-content-center",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Hide",
          "type": "button",
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Col
          lg={
            Object {
              "offset": 3,
              "size": 6,
            }
          }
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <RandomChar />
          <button
            className="button-hide d-flex justify-content-center"
            onClick={[Function]}
          >
            Hide
          </button>
        </Col>,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <RandomChar />,
            <button
              className="button-hide d-flex justify-content-center"
              onClick={[Function]}
            >
              Hide
            </button>,
          ],
          "lg": Object {
            "offset": 3,
            "size": 6,
          },
          "tag": "div",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Hide",
              "className": "button-hide d-flex justify-content-center",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Hide",
            "type": "button",
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
